ouroboros-consensus/src/Ouroboros/Consensus/Block/Abstract.hs

```diff
@@ -131,25 +133,43 @@ headerPoint = castPoint . blockPoint
 -------------------------------------------------------------------------------}
 
 -- | Convert a hash from/to raw bytes
+--
+-- Variants of 'toRawHash' and 'fromRawHash' for 'ShortByteString' are
+-- provided. Override the default implementations to avoid an extra step in
+-- case the 'HeaderHash' is a 'ShortByteString' under the hood.
 class ConvertRawHash blk where
   -- | Get the raw bytes from a hash
-  toRawHash   :: proxy blk -> HeaderHash blk -> Strict.ByteString
+  toRawHash :: proxy blk -> HeaderHash blk -> Strict.ByteString
+  toRawHash p = Short.fromShort . toShortRawHash p
 
   -- | Construct the hash from a raw hash
   --
   -- PRECONDITION: the bytestring's size must match 'hashSize'
   fromRawHash :: proxy blk -> Strict.ByteString -> HeaderHash blk
+  fromRawHash p = fromShortRawHash p . Short.toShort
+
+  -- | Variant of 'toRawHash' for 'ShortByteString'
+  toShortRawHash :: proxy blk -> HeaderHash blk -> ShortByteString
+  toShortRawHash p = Short.toShort . toRawHash p
+
+  -- | Variant of 'fromRawHash' for 'ShortByteString'
+  fromShortRawHash :: proxy blk -> ShortByteString -> HeaderHash blk
+  fromShortRawHash p = fromRawHash p . Short.fromShort
 
   -- | The size of the hash in number of bytes
   hashSize    :: proxy blk -> Word32
 
+  {-# MINIMAL hashSize
+            , (toRawHash | toShortRawHash)
+            , (fromRawHash | fromShortRawHash) #-}
+
 encodeRawHash :: ConvertRawHash blk
               => proxy blk -> HeaderHash blk -> Encoding
-encodeRawHash p = Serialise.encode . toRawHash p
+encodeRawHash p = Serialise.encode . toShortRawHash p
```

Comment:
When/if we decide to use `ShortByteString` for our hashes, we should audit consensus for other uses of `toRawHash` and `fromRawHash` that should be replaced with their `ShortByteString` variants.

Reply:
Done